core/parsigdb/memory.go

```diff
@@ -167,7 +214,12 @@ func (db *MemDB) store(k key, value core.ParSignedData) ([]core.ParSignedData, b
 	}
 
 	db.entries[k] = append(db.entries[k], clone)
-	db.keysByDuty[k.Duty] = append(db.keysByDuty[k.Duty], k)
+
+	// Only append key to keysByDuty if this is the first signature for this key.
+	// This prevents duplicate keys accumulating in keysByDuty.
+	if len(db.entries[k]) == 1 {
+		db.keysByDuty[k.Duty] = append(db.keysByDuty[k.Duty], k)
```

Comment:
To be fair, I can't see where this `keysByDuty` is read at all? I see only writes and deletes. Can we... get rid of it?